libcontainer/system/linux.go

```diff
@@ -215,3 +230,103 @@ func SetLinuxPersonality(personality int) error {
 	}
 	return nil
 }
+
+// UpdateSetgroups is to set the process's setgroups policy
+// This *must* be called before we touch gid_map.
+func UpdateSetgroups(pid int, policy SetgroupsPolicy) error {
+	var strPolicy string
+	switch policy {
+	case SetgroupsAllow:
+		strPolicy = "allow"
+	case SetgroupsDeny:
+		strPolicy = "deny"
+	case SetgroupsDefault:
+		fallthrough
+	default:
+		return nil
+	}
+	err := utils.WriteString(fmt.Sprintf("/proc/%d/setgroups", pid), strPolicy)
```

Comment:
In general, Go's Sprintf is slow (way slower than in C), so it's better to do something like this:
```go
file := "/proc/" + strconv.Itoa(pid) + "/setgroups"
```

(you could also use filepath.Join instead of `+` but that doesn't add any value for constant paths).